File: backend/src/main/java/todoktodok/backend/comment/application/dto/CommentResponse.java
package todoktodok.backend.comment.application.dto;

import java.time.LocalDateTime;
import todoktodok.backend.comment.domain.Comment;
import todoktodok.backend.member.application.dto.response.MemberResponse;

public record CommentResponse(
        Long commentId,
        MemberResponse member,
        LocalDateTime createdAt,
        String content
) {

    public CommentResponse(final Comment comment){
        this(
                comment.getId(),
                new MemberResponse(comment.getMember()),
                comment.getCreatedAt(),
                comment.getContent()
        );
    }
}

File: android/TodokTodok/app/src/main/java/com/example/todoktodok/data/network/response/comment/CommentResponse.kt
package com.example.todoktodok.data.network.response.comment

import com.example.domain.model.Comment
import com.example.todoktodok.data.core.ext.toLocalDateTime
import com.example.todoktodok.data.network.response.discussion.MemberResponse
import com.example.todoktodok.data.network.response.discussion.toDomain
import kotlinx.serialization.Serializable

@Serializable
data class CommentResponse(
    val commentId: Long,
    val content: String,
    val createdAt: String,
    val member: MemberResponse,
)

fun CommentResponse.toDomain() = Comment(commentId, content, member.toDomain(), createdAt.toLocalDateTime())
